test/e2e/velero_utils.go

```diff
@@ -386,10 +386,8 @@ func waitForVSphereUploadCompletion(ctx context.Context, timeout time.Duration,
 			// Canceled - the operation was canceled, the snapshot ID is not valid
 			if len(comps) == 2 {
 				phase := comps[1]
-				if phase == "New" ||
-					phase == "InProgress" ||
-					phase == "Snapshotted" ||
-					phase == "Uploading" {
+				// if the phase isn't "Uploaded", the uploading is either in progress or failed(will retry, if the retrying cannot recover, will timeout in the final)
+				if phase != "Uploaded" {
```

Comment:
We should return an error in the case where the snapshot reached a failed terminal state as it's not possible to recover from that. By checking for `Uploaded` only, this function could poll for up to an hour for a snapshot that we know can't be used.

Reply:
@zubron I noticed that seems the uploading will be retry by the vSphere plugin, I'm not sure whether the failed uploading can be fixed by the retrying, so don't return the error here

Should we expect the retrying may fix the failed uploading or just return the error?

Reply:
Oh, sorry - I didn't realize that it would retry the upload :woman_facepalming: In that case, it's probably best to ignore my comment and check with @dsu-igeek :)